File: files/CH-218.R
library(tidyverse)
library(readxl)

path = "files/CH-218 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:E7")

result = input %>%
  mutate(
    Uppercase = map_chr(str_extract_all(ID, "[A-Z]+"), ~ paste(.x, collapse = "")),
    Lowercase = map_chr(str_extract_all(ID, "[a-z]+"), ~ paste(.x, collapse = "")))

File: files/200-299/218/CH-218.py
import pandas as pd
import re

path = "CH-218 Column Splitting.xlsx"
input_data = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test_data = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=6)

input_data["Uppercase"] = input_data["ID"].apply(lambda x: "".join(re.findall(r"[A-Z]+", x)))
input_data["Lowercase"] = input_data["ID"].apply(lambda x: "".join(re.findall(r"[a-z]+", x)))

print(input_data)